daphne_worker/src/dap.rs

```diff
@@ -363,57 +377,18 @@ where
         Ok(None)
     }
 
-    /// Get an existing task (whether an ordinary task or a previously created
-    /// taskprov task).  If we can't find it, see if there is a taskprov extension
-    /// in the report, and if so create the task.
-    async fn resolve_taskprov(
+    async fn taskprov_put(
         &'srv self,
-        task_id: &TaskId,
         req: &'req DapRequest<DaphneWorkerAuth>,
-        report_metadata_advertisement: Option<&ReportMetadata>,
+        task_config: DapTaskConfig,
     ) -> std::result::Result<(), DapError> {
-        let found = self
-            .get_task_config(Cow::Borrowed(task_id))
-            .await
-            .map_err(dap_err)?;
-        if found.is_some() {
-            return Ok(());
-        }
-
-        let global_config = self.get_global_config();
-        if !global_config.allow_taskprov {
-            // TODO(bhalleycf) if DAP gets a generic denied error, we should use it here.
-            return Err(DapError::Abort(DapAbort::InvalidTask {
-                detail: "Taskprov extension is disabled.".to_string(),
-                task_id: task_id.clone(),
-            }));
```

Comment:
I don't think aborting here is actually desirable. If taskprov is disabled then we just want to ensure we don't try to configure a task.